client/src/components/Form/LoginForm.js

```diff
@@ -1,49 +1,71 @@
 import React, { useState } from "react";
 import { useForm } from "react-hook-form";
 import { useAuth } from "../../hooks/useAuth";
-import { Button, TextField, Typography } from "@material-ui/core";
+import { Button, TextField, Typography, makeStyles } from "@material-ui/core";
 import Alert from "@material-ui/lab/Alert";
-
 import Form from "./Form";
+import * as yup from "yup";
+import { yupResolver } from "@hookform/resolvers/yup";
+
+const schema = yup.object().shape({
+  username: yup.string().required("A username is required"),
+  password: yup.string().required("Password is required"),
+});
+
+const useStyles = makeStyles((theme) => ({
+  input: theme.input,
+  formButton: theme.formButton,
+  error: theme.error,
+}));
 
-export default function LoginForm({ classes }) {
+export default function LoginForm() {
   const [error, setError] = useState(null);
-  const { register, handleSubmit, errors } = useForm();
+  const { register, handleSubmit, errors } = useForm({
+    resolver: yupResolver(schema),
+  });
 
+  const classes = useStyles();
   const auth = useAuth();
 
   const onSubmit = async (data) => {
     const response = await auth.login(data);
-    if (response) {
+    console.log(response);
+    if (response.user) {
       setError(false);
-      // continue to next page
-      alert("logged in.");
+      console.log("success");
+      // move forward
+    } else if (response.msg) {
+      setError({ msg: response.msg });
```

Comment:
To prevent any confusion, it might be better to send error messages under an `error` key rather than the `msg` key if there are any errors. That way you could also send a success message using the `msg` key if necessary.